File: backend/models/logging.py
from enum import Enum


class Subsystem(str, Enum):
    """Backend subsystem identifiers for logging"""
    # Core application
    APP = "app"
    API = "api"
    
    # Services
    MEMORY = "memory"
    PAYMENT = "payment"
    
    # External integrations
    AGENT_COMM = "agent-comm"
    DATABASE = "database"
    
    # Infrastructure
    HEALTH = "health"
    UTILS = "utils"


class ErrorType(str, Enum):
    """Error type classifications for structured logging"""
    # Database errors
    DATABASE_CONNECTION = "databaseConnection"
    DATABASE_QUERY = "databaseQuery"
    
    # Agent communication errors
    AGENT_TIMEOUT = "agentTimeout"
    AGENT_UNAVAILABLE = "agentUnavailable"
    INVALID_AGENT_RESPONSE = "invalidAgentResponse"
    
    # Payment errors
    PAYMENT_FAILED = "paymentFailed"
    PAYMENT_INVALID = "paymentInvalid"
    
    # API errors
    INVALID_REQUEST = "invalidRequest"
    AUTHENTICATION_FAILED = "authenticationFailed"
    RATE_LIMITED = "rateLimited"
    
    # Infrastructure errors
    NETWORK_ERROR = "networkError"
    TIMEOUT = "timeout"
    CONFIGURATION_ERROR = "configurationError" 

File: frontend/src/services/types/logging.ts
export enum Subsystem {
  // Core application
  APP = 'app',
  CHAT = 'chat',
  
  // Components
  SIDEBAR = 'sidebar',
  INPUT = 'input',
  MESSAGES = 'messages',
  
  // Services  
  AGENT_COMM = 'agent-comm',
  STORAGE = 'storage',
  WALLET = 'wallet',
  
  // External integrations
  POLKADOT_API = 'polkadot-api',
  SOCKET_IO = 'socket-io',
  
  // Infrastructure
  UTILS = 'utils',

  // TODO: Rethink subsystems
}

export enum ErrorType {
  // Agent communication errors
  AGENT_TIMEOUT = 'agentTimeout',
  AGENT_UNAVAILABLE = 'agentUnavailable',
  INVALID_AGENT_RESPONSE = 'invalidAgentResponse',
  
  // Wallet errors
  WALLET_NOT_CONNECTED = 'walletNotConnected',
  WALLET_REJECTED = 'walletRejected',
  INSUFFICIENT_BALANCE = 'insufficientBalance',
  
  // Network errors  
  NETWORK_ERROR = 'networkError',
  RPC_ERROR = 'rpcError',
  CONNECTION_FAILED = 'connectionFailed',
  
  // Storage errors
  STORAGE_FAILED = 'storageFailed',
  
  // UI/UX errors
  INVALID_INPUT = 'invalidInput',
  COMPONENT_ERROR = 'componentError',

  // TODO: Rethink error types
}

